dynosaur_derive/src/expand.rs

```diff
@@ -148,6 +135,14 @@ pub(crate) fn expand_sig_ret_ty_to_pin_box(sig: &mut Signature) {
     sig.output = parse_quote! { #arrow ::core::pin::Pin<Box<dyn #ret + 'dynosaur>> };
 }
 
+pub(crate) fn expand_sig_ret_ty_to_box(sig: &mut Signature) {
+    let (arrow, ret) = expand_arrow_ret_ty(sig);
+    if let Some(asyncness) = sig.asyncness.take() {
+        sig.fn_token.span = asyncness.span;
+    }
+    sig.output = parse_quote! { #arrow Box<dyn #ret + '_> };
```

Comment:
The lifetime should work the same as in the pin_box case. If there's a single lifetime as in the current test we can simplify it all to `'_`, but otherwise we need the full `'dynosaur` thing. The return type can capture all input lifetimes, we can't assume it captures only the self lifetime (which `'_` would do).. we have to be very careful here because the unsafe lifetime extension is not checked by the compiler.